src/napari_spatialdata/_interactive.py

```diff
@@ -1,203 +1,12 @@
-
-    def _add_polygons(self, key: str) -> None:
-        polygons = []
-        df = self._sdata.shapes[key]
-        affine = _get_transform(self._sdata.shapes[key], self.coordinate_system_widget._system)
-
-        # when mulitpolygons are present, we select the largest ones
-        if "MultiPolygon" in np.unique(df.geometry.type):
-            logger.info("Multipolygons are present in the data. Only the largest polygon per cell is retained.")
-            df = df.explode(index_parts=False)
-            df["area"] = df.area
-            df = df.sort_values(by="area", ascending=False)  # sort by area
-            df = df[~df.index.duplicated(keep="first")]  # only keep the largest area
-            df = df.sort_index()  # reset the index to the first order
-        if len(df) < 100:
-            for i in range(0, len(df)):
-                polygons.append(list(df.geometry.iloc[i].exterior.coords))
-        else:
-            for i in range(
-                0, len(df)
-            ):  # This can be removed once napari is sped up in the plotting. It changes the shapes only very slightly
-                polygons.append(list(df.geometry.iloc[i].exterior.simplify(tolerance=2).coords))
-        # this will only work for polygons and not for multipolygons
-        polygons = _swap_coordinates(polygons)
-
-        self._viewer.add_shapes(
-            polygons,
-            name=key,
-            affine=affine,
-            shape_type="polygon",
-            metadata={
-                "adata": self._sdata.table[
-                    self._sdata.table.obs[self._sdata.table.uns["spatialdata_attrs"]["region_key"]] == key
-                ],
-                "shapes_key": self._sdata.table.uns["spatialdata_attrs"]["region_key"],
-                "shapes_type": "polygons",
-            },
-        )
-
-    def _add_shapes(self, key: str) -> None:
-        if type(self._sdata.shapes[key].iloc[0][0]) == shapely.geometry.point.Point:
-            self._add_circles(key)
-        elif (type(self._sdata.shapes[key].iloc[0][0]) == shapely.geometry.polygon.Polygon) or (
-            type(self._sdata.shapes[key].iloc[0][0]) == shapely.geometry.multipolygon.MultiPolygon
-        ):
-            self._add_polygons(key)
-        else:
-            raise TypeError(
-                "Incorrect data type passed for shapes (should be Shapely Point or Polygon or MultiPolygon)."
-            )
-
-    def _add_label(self, key: str) -> None:
-        affine = _get_transform(self._sdata.labels[key], self.coordinate_system_widget._system)
-
-        self._viewer.add_labels(
-            self._sdata.labels[key],
-            name=key,
-            affine=affine,
-            metadata={
-                "adata": self._sdata.table[
-                    self._sdata.table.obs[self._sdata.table.uns["spatialdata_attrs"]["region_key"]] == key
-                ],
-                "labels_key": self._sdata.table.uns["spatialdata_attrs"]["instance_key"],
-            },
-        )
```

Comment:
this part is replaced by the big/complex function below. I think this way to handle annotation metadata is much cleaner and would keep this.